components/kyma-environment-broker/internal/broker/server.go

```diff
@@ -0,0 +1,52 @@
+package broker
+
+import (
+	"code.cloudfoundry.org/lager"
+	"github.com/gorilla/mux"
+	"github.com/pivotal-cf/brokerapi/v7/auth"
+	"github.com/pivotal-cf/brokerapi/v7/domain"
+	"github.com/pivotal-cf/brokerapi/v7/handlers"
+	"github.com/pivotal-cf/brokerapi/v7/middlewares"
+)
+
+type BrokerCredentials struct {
+	Username string
+	Password string
+}
+
+// copied from github.com/pivotal-cf/brokerapi/api.go
+func New(serviceBroker domain.ServiceBroker, logger lager.Logger, brokerCredentials *BrokerCredentials) *mux.Router {
+	router := mux.NewRouter()
+
+	AttachRoutes(router, serviceBroker, logger)
+
+	if brokerCredentials != nil {
```

Comment:
the middlewares are added on stack and right now you are changing the default behavior the `router.Use(middlewares.AddCorrelationIDToContext)` should be before the auth

Reply:
Fixed